### PR TITLE
Refactor: Simplize the way of creating paral_config from json file in config tuner.

### PR DESCRIPTION
**Pull request by matmua**
### What changes were proposed in this pull request?

This modification enhances the process of creating a class instance through JSON files, making the JSON reading process more versatile and generic.

### Why are the changes needed?

The previous functionality was achieved through a simpler approach, which required modifying this section of code every time there was a change in classes. With the optimization, it has been automated.

### Does this PR introduce any user-facing change?

No, It only enhances the extensibility of this module.

### How was this patch tested?

This patch hasn't been tested.

### CODE REVIEW

File: dlrover/python/elastic_agent/config/paral_config_tuner.py [hunk 1]
```diff
@@ -75,28 +75,16 @@ def _read_paral_config(self, config_path):
...
-                        version=config_data["optimizer"]["version"],
-                        optimizer_name=config_data["optimizer"][
-                            "optimizer_name"
-                        ],
-                        learning_rate=config_data["optimizer"][
-                            "learning_rate"
-                        ],
-                    ),
-                )
+            with open(config_path, 'r') as json_file:
```
Comment (Antlera) [inline dlrover/python/elastic_agent/config/paral_config_tuner.py:78]: We should check the config file whether is empty. 
```python
    try:
         content = f.read()
         config = json.loads(content)
         except json.decoder.JSONDecodeError:
              logger.warning(f"Fail to load config {content}")
              return
```

Comment (Antlera) [reply dlrover/python/elastic_agent/config/paral_config_tuner.py:78]: This is a corner case.

Comment (matmua) [reply dlrover/python/elastic_agent/config/paral_config_tuner.py:78]: have solved in the second commit.

**Comment by Antlera** [comment]
LGTM

**Review by Antlera — APPROVED** [review]

**Comment by Antlera** [comment]
This pr fix response to #700

File: dlrover/python/elastic_agent/config/paral_config_tuner.py [hunk 2]
```diff
@@ -76,27 +76,18 @@ def _read_paral_config(self, config_path):
...
-                        optimizer_name=config_data["optimizer"][
-                            "optimizer_name"
-                        ],
-                        learning_rate=config_data["optimizer"][
-                            "learning_rate"
-                        ],
-                    ),
-                )
+                content = json_file.read()
+                json_data = json.loads(content)
```
Comment (workingloong) [inline dlrover/python/elastic_agent/config/paral_config_tuner.py:80]: Check whether the content is an empty string before loading it.

```python
if not content:
    return self.config.

```

File: dlrover/python/elastic_agent/config/paral_config_tuner.py [hunk 3]
```diff
@@ -76,27 +76,18 @@ def _read_paral_config(self, config_path):
...
+            # if we find the instance_name in the JSON file, create it
+            items = json_data.items()
+            for item in items:
```
Comment (workingloong) [inline dlrover/python/elastic_agent/config/paral_config_tuner.py:83]: line 82 and 83 can be simplified a line

```python
for key, value in json_data.items():
    if key == "dataloader":
        dataloader = DataloaderConfig(**value)
   elif key == ...
```

**Comment by merlintang** [comment]
LGTM

**Review by merlintang — COMMENTED** [review]
LGTM

File: dlrover/python/elastic_agent/config/paral_config_tuner.py (hunk 2 above)
Comment (matmua) [reply dlrover/python/elastic_agent/config/paral_config_tuner.py:80]: this situation has been checked by:
        except json.JSONDecodeError as e:
            print(f"Error: Failed to decode JSON file '{config_path}': {e}")
            return None
They are at the 95-97 line.

File: dlrover/python/elastic_agent/config/paral_config_tuner.py (hunk 3 above)
Comment (matmua) [reply dlrover/python/elastic_agent/config/paral_config_tuner.py]: New codes have been pushed.

**Review by workingloong — APPROVED** [review]
LGTM

**Review by workingloong — APPROVED** [review]
LGTM

**Pull request merged**